code/modules/mob/living/simple_animal/constructs/constructs.dm

```diff
@@ -217,19 +217,28 @@
 	var/reflectchance = 80 - round(P.damage/3)
 	if(prob(reflectchance))
 		var/damage_mod = rand(2,4)
+		var/projectile_dam_type = P.damage_type
+		var/incoming_damage = (round(P.damage / damage_mod) - (round((P.damage / damage_mod) * 0.3)))
+		var/armorcheck = run_armor_check(null, P.check_armour)
+		var/soakedcheck = get_armor_soak(null, P.check_armour)
 		if(!(istype(P, /obj/item/projectile/energy) || istype(P, /obj/item/projectile/beam)))
 			visible_message("<span class='danger'>The [P.name] bounces off of [src]'s shell!</span>", \
 						"<span class='userdanger'>The [P.name] bounces off of [src]'s shell!</span>")
 			new /obj/item/weapon/material/shard/shrapnel(src.loc)
-			var/incoming_brute = (round(P.damage / damage_mod) - (round((P.damage / damage_mod) * 0.3))) //Calculates incoming damage with respect to standard resistance.
-			adjustBruteLoss(incoming_brute)
+			if(P.damage_type != BRUTE || P.damage_type != BURN)
```

Comment:
Just have to check that it isn't the special case (BURN), the default case (BRUTE) will still default to brute

Reply:
Brute beams swapped to burn, so I've got a swap for it to correct it. Not.. entirely certain why the new one works and the other doesn't.